File: src/feature/handling/handle_rc.rb
# frozen_string_literal: true

require_relative '../../model/rice_cooker'

# Module RcHandler containing actions on rice cooker's list
module RcHandler
  @rc_list = []

  def self.add(id, is_operational)
    new_rc = RiceCooker.new(id, is_operational)
    @rc_list.push(new_rc)
    puts "New rice cooker with id: #{id} successfully added."
  end

  def self.change_state(id, target_attr, state)
    target_rc = @rc_list.find { |rc| rc.id == id }
    if target_rc
      case target_attr
      when 'is_operational'
        target_rc.is_operational = state
      when 'is_cooking'
        target_rc.is_cooking = state
      when 'is_plugged'
        target_rc.is_plugged = state
      else
        puts 'The targeted attribute is not valid'
      end
    else
      puts "The rice cooker with id: #{id} doesn't exist."
    end
  end

  def self.rc_list
    @rc_list
  end

  def self.get_rc(id)
    found = @rc_list.find { |rc| rc.id == id }
    return found if found

    puts "The rice cooker with id: #{id} doesn't exist."
  end
end

File: app/src/main/kotlin/my/rice/cooker/feature/HandleRc.kt
val rcList = mutableListOf<RiceCooker>()

fun add(id: Int, isOperational: Boolean) {
    val newRC = RiceCooker(id, isOperational)
    rcList.add(newRC)
    println("New rice cooker with id: $id successfully added.")
}

fun changeState(id: Int, targetAttr: String, state: Boolean) {
    val targetRC = rcList.find { it.id == id }
    if (targetRC != null) {
        when (targetAttr) {
            "isOperational" -> targetRC.isOperational = state
            "isCooking" -> targetRC.isCooking = state
            "isPlugged" -> targetRC.isPlugged = state
            else -> println("The targeted attribute is not valid")
        }
    } else {
        println("The rice cooker with id: $id doesn't exist.")
    }
}

fun getRCList(): List<RiceCooker> {
    return rcList.toList()
}

fun getRC(id: Int): RiceCooker? {
    return rcList.find { it.id == id }
}